tabular/src/autogluon/tabular/models/automm/automm_model.py

```diff
@@ -127,8 +129,11 @@ def _fit(self,
             logger.log(15, "sample_weight not yet supported for MultiModalPredictorModel, "
                            "this model will ignore them in training.")
 
+        # Need to deep copy to avoid altering outer context
+        X_train = copy.deepcopy(X_train)
```

Comment:
We can use `X_train.copy()`. But the current logic looks good.

Reply:
good point, updated